cli/tests/activate.bats

```diff
@@ -924,3 +924,76 @@ EOF
 }
 
 # ---------------------------------------------------------------------------- #
+
+# bats test_tags=activate,activate:custom_zdotdir,activate:custom_zdotdir:bash
+@test "bash: preserve custom ZDOTDIR" {
+  "$FLOX_BIN" delete -f
+  "$FLOX_BIN" init
+  FLOX_SHELL=bash ZDOTDIR=/custom/zdotdir run "$FLOX_BIN" activate -- echo '$ZDOTDIR'
+  assert_success
+  assert_line "/custom/zdotdir"
+}
+
+# bats test_tags=activate,activate:custom_zdotdir,activate:custom_zdotdir:zsh
+@test "zsh: preserve custom ZDOTDIR" {
+  "$FLOX_BIN" delete -f
+  "$FLOX_BIN" init
+  FLOX_SHELL=zsh ZDOTDIR=/custom/zdotdir run "$FLOX_BIN" activate -- echo '$ZDOTDIR'
+  assert_success
+  assert_line "/custom/zdotdir"
+}
+
+# ---------------------------------------------------------------------------- #
+
+# bats test_tags=activate,activate:zdotdir,activate:zdotdir:zshenv
+@test "zdotdir: test zshenv activation" {
+  echo "echo sourcing .zshenv" > "$HOME/.zshenv"
+  echo "echo sourcing .zshrc" > "$HOME/.zshrc"
+  echo "echo sourcing .zlogin" > "$HOME/.zlogin"
+  "$FLOX_BIN" delete -f
+  "$FLOX_BIN" init
+  run "$FLOX_BIN" install hello
```

Comment:
nonblocking: do we need the `install`? That just makes the test slower

Reply:
yeah I was looking for some other way to do that; there's no sentinel variable set by the zsh init script, the best I can imagine is looking for `$FLOX_ENV/bin` as the first element in the `$PATH`. If you want to make that change then it's fine by me, or we could add a sentinel variable to each of the `./pkgdb/src/buildenv/assets/activate.d/{bash,zsh}` init scripts?

Reply:
Added patch to fix this in https://github.com/flox/flox/pull/1475